### PR TITLE
[dv,py] Tweak ISS linker arg construction for Xcelium

### LINKED ISSUE #2205: Compile_tb.py generate options not recogised by xrun
## Observed Behavior

[compile_tb_stdstreams.log](https://github.com/user-attachments/files/16634707/compile_tb_stdstreams.log)

## My Environment

**EDA tool and version:**
 Xcelium 23.09-s001

**Operating system:**
 Centos 7.9

**Version of the Ibex source code:**
 668233699df9ec2a40413e69e0de0a5b10185980

### PR DESCRIPTION
**Pull request by hcallahan-lowrisc**
The previous code here was a bit too hacky, so implement a solution that directly follows the suggestion in the Cadence support article. An example was also added to make it clear what this transformation is achieving.

Additionally, cleanup some of the other code in the module, renaming for clarity and adding comments. Move some variable declarations around to be closer to their use. Add some more typehints.

Fixes #2205

### CODE REVIEW

**Review by rswarbrick — COMMENTED** [review]
This drops the

File: dv/uvm/core_ibex/scripts/compile_tb.py [hunk 1]
```diff
@@ -19,23 +21,51 @@
...
-        # the following hack to get it through.
-        if '-Wl' in _flags:  # This should be in LDFLAGS only
-            _flags = "'-Xlinker {}'".format(_flags.replace('-Wl,', ''))
-    return _flags
+    # Generate a list of tokens for each call, and append it to the all_tokens variable
+    for s in specifiers:
+        cmd = ['pkg-config', s] + iss_pc
+        stdout = subprocess.check_output(cmd, universal_newlines=True).strip()
+        tokens = shlex.split(stdout)
+        logger.debug(f"pkgconfig_tokens = {tokens}")
```
Comment (rswarbrick) [inline dv/uvm/core_ibex/scripts/compile_tb.py:34]: I'm not sure I know enough about Python logging to have an opinion about this: do you want to keep these debug prints?

File: dv/uvm/core_ibex/scripts/compile_tb.py [hunk 2]
```diff
@@ -19,23 +21,51 @@
...
+
+        rpath_prefix = '-Wl,-rpath,'
+        def fixup_xcelium_rpath_token(t: str) -> str:
+            """Re-format rpath flags to ensure reliable passing to Xcelium.
+
+            When passing rpath flags to xcelium through the xrun tool, we need to re-format the string
+            output by pkg-config to ensure reliability.
```
Comment (rswarbrick) [inline dv/uvm/core_ibex/scripts/compile_tb.py:41]: Nit: I think you probably want "output **from** pkg-config" here.

**Review by rswarbrick — APPROVED** [review]
Looks sensible to me!

**Comment by hcallahan-lowrisc** [comment]
Thanks!

**Pull request merged**